libqfieldsync/offline_converter.py

```diff
@@ -346,6 +350,18 @@ def _convert(self, project: QgsProject) -> None:
         QgsProject.instance().write(str(export_project_filename))
         project.writeProject.disconnect(on_original_project_write)
 
+    def remove_empty_groups_from_project(self, group) -> None:
+        """
+        Recursively removes any empty groups from the given layer tree group.
+        """
+        for child in group.children():
+            if isinstance(child, QgsLayerTreeGroup):
+                self.remove_empty_groups_from_project(child)
+
+                if not child.children():
+                    print(f"Removing empty group: {child.name()}")
+                    group.removeChildNode(child)
```

Comment:
```suggestion
        for child in group.children():
            if not isinstance(child, QgsLayerTreeGroup):
                continue

            self.remove_empty_groups_from_project(child)

            if not child.children():
                print(f"Removing empty group: {child.name()}")
                group.removeChildNode(child)
```

This pattern is called "early return", or in our case, "early continue". It makes the code less nested and shows clear intention what kind of context you want to manage. You can find it across QFieldSync a lot, so whenever possible and makes sense - try to use it :)

Reply:
Great, indeed is much better!